instance/models/instance.py

```diff
@@ -134,6 +152,68 @@ def status(self):
         else:
             return active_server_set[0].status
 
+    @property
+    def event_context(self):
+        """
+        Context dictionary to include in events
+        """
+        return {'instance_id': self.pk}
+
+    @staticmethod
+    def on_post_save(sender, instance, created, **kwargs):
+        """
+        Triggered by the post_save event, with `created` a boolean indicating if the object was just
+        created in the DB
+        """
+        self = instance
+        self.update_logger()
+
+    def update_logger(self):
+        """
+        Start referencing the instance in logs once the DB entry exist
+        """
+        if self.pk is not None:
+            self.logger = InstanceLoggerAdapter(logger, {'obj': self})
```

Comment:
@antoviaque Having to call `update_logger` feels dirty to me. I would prefer if `self.logger` was always created in the `__init__` method, with `'obj'` set to `self`, regardless of whether the object has been saved to the database yet. The `if self.pk is not None` checks could be moved to the logger adapters instead (`DBHandler` already does that).